capellambse_context_diagrams/collectors/default.py

```diff
@@ -274,8 +274,14 @@ def port_exchange_collector(
     edges: dict[str, common.ElementList[fa.AbstractExchange]] = {}
     for i in ports:
         try:
-            if edge := filter(getattr(i, "exchanges")):
-                edges[i.uuid] = edge
+            if exs := filter(getattr(i, "exchanges")):
+                edges[i.uuid] = exs
+                continue
+        except AttributeError:
+            pass
+        try:
+            if links := filter(getattr(i, "links")):
+                edges[i.uuid] = links
         except AttributeError:
             pass
```

Comment:
I don't like the portion of the code. I'd change the following for more readability:
```
def _extract_edges(
    obj: common.ElementList[common.GenericElement],
    attribute: str,
    filter: cabc.Callable[
        [cabc.Iterable[common.GenericElement]],
        cabc.Iterable[common.GenericElement],
    ],
) -> common.ElementList[common.GenericElement] | list:
    return filter(getattr(obj, attribute, []))


for port in ports:
    if exs := _extract_edges(port, "exchanges", filter):
        edges[port.uuid] = exs
        continue

    if links := _extract_edges(port, "links", filter):
        edges[port.uuid] = links
```

Then please define a type alias for the filter type.